openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -32,59 +32,65 @@ function getJqueryElements(selector){
-                if (data.value == '') {
-                    return error('#id-errors', 'id-value', dataConfig['You need to give a value to ID.'].replace(/ID/, label));
-                }
-                if (['ocaid'].includes(data.name) && /\s/g.test(data.value)) {
-                    return error('#id-errors', 'id-value', dataConfig['ID ids cannot contain whitespace.'].replace(/ID/, label));
-                }
-                if (data.name == 'isbn_10' && data.value.length != 10) {
-                    return error('#id-errors', 'id-value', dataConfig['ID must be exactly 10 characters [0-9] or X.'].replace(/ID/, label));
-                }
-                if (data.name == 'isbn_13' && data.value.replace(/-/g, '').length != 13) {
-                    return error('#id-errors', 'id-value', dataConfig['ID must be exactly 13 digits [0-9]. For example: 978-1-56619-909-4'].replace(/ID/, label));
-                }
-                $('id-errors').hide();
-                return true;
+            if (data.name == '') {
+                return error('#role-errors', '#role-name', dataConfig['You need to give this ROLE a name.'].replace(/ROLE/, data.role));
             }
-        });
-        $('#classifications').repeat({
-            vars: {prefix: 'edition--'},
-            validate: function (data) {
-                if (data.name == '' || data.name == '---') {
-                    return error('#classification-errors', '#select-classification', dataConfig['Please select a classification.']);
-                }
-                if (data.value == '') {
-                    const label = $('#select-classification').find(`option[value=${data.name}]`).html();
-                    return error('#classification-errors', '#classification-value', dataConfig['You need to give a value to CLASS.'].replace(/CLASS/, label));
-                }
-                $('#classification-errors').hide();
-                return true;
+            $('#role-errors').hide();
+            return true;
+        }
+    });
+}
+
+export function initIdentifierValidation() {
+    const dataConfig = JSON.parse(document.querySelector('#identifiers').dataset.config);
+    $('#identifiers').repeat({
+        vars: {prefix: 'edition--'},
+        validate: function (data) {
+            if (data.name == '' || data.name == '---') {
+                return error('#id-errors', 'select-id', dataConfig['Please select an identifier.'])
             }
-        });
+            const label = $('#select-id').find(`option[value=${data.name}]`).html();
+            if (data.value == '') {
+                return error('#id-errors', 'id-value', dataConfig['You need to give a value to ID.'].replace(/ID/, label));
+            }
+            if (['ocaid'].includes(data.name) && /\s/g.test(data.value)) {
+                return error('#id-errors', 'id-value', dataConfig['ID ids cannot contain whitespace.'].replace(/ID/, label));
```

Comment:
I recommend using a better key then 'ID ids cannot contain whitespace.'
e.g. whitespaceError.

Reply:
This is the pattern that Drini recommended for dealing with translations until we have a more robust system to support them on the JS side. I agree it's not pretty. Lets come to a decision on how to handle i18n on the js side of things and then change all of them at once.

Reply:
See above ^.

Presumably, when we have a more robust system we'll need to replace all this code anyhow. Having readable JavaScript should make that more easier to grok.